R/preprocessReads.R

```diff
@@ -1,363 +1,567 @@
             
-                mode <- 'a'
-            }
-
-        } else {
-            stop("unknown file format: ", fileformat)
+            ## truncate start or end bases
+            chunks <- ShortRead::narrow(x = chunks, start = truncateFromBase, 
+                                        end = truncateToBase)
+            
+            ## trim adaptor
+            ranges <- Biostrings::trimLRPatterns(
+                subject = chunks, Lpattern = Lpattern, Rpattern = Rpattern, 
+                max.Lmismatch = max.Lmismatch, max.Rmismatch = max.Rmismatch,
+                with.Lindels = with.Lindels, with.Rindels = with.Rindels, 
+                ranges = TRUE
+            )
+            filterReport['matchTo5pAdapter'] <- filterReport['matchTo5pAdapter'] + 
+                sum(BiocGenerics::start(ranges) != 1)
+            filterReport['matchTo3pAdapter'] <- filterReport['matchTo3pAdapter'] + 
+                sum(BiocGenerics::end(ranges) != BiocGenerics::width(chunks))
+            chunks <- ShortRead::narrow(x = chunks, 
+                                        start = BiocGenerics::start(ranges), 
+                                        end = BiocGenerics::end(ranges))
+        
+            ## filter and write short reads
+            filterResults <- S4Vectors::evalSeparately(filters, chunks)
+            #filterReport['tooManyN'] <- filterReport['tooManyN'] + sum(!filterResults[, 'CleanNFilter'])
+            # workaround for filter being renamed to 'CleanNFilter.other' if nrec=1:
+            filterReport['tooManyN'] <- filterReport['tooManyN'] + 
+                sum(!filterResults[, grep('CleanNFilter',colnames(filterResults))[1]])
+            filterReport['tooShort'] <- filterReport['tooShort'] + sum(!filterResults[, 'LengthFilter'])
+            filterReport['lowComplexity'] <- filterReport['lowComplexity'] + 
+                sum(!filterResults[, 'LowComplexityFilter'])
+            filter <- apply(filterResults, 1, all)
+            filterReport['totalPassed'] <- filterReport['totalPassed'] + sum(filter)
+            if (sum(filter))
+                ShortRead::writeFastq(chunks[filter], tmpOutputFilename, 
+                                      mode = mode, qualityType = "Auto", compress = FALSE)
+            mode <- 'a'
         }
-
-        if(filecompr != "none")
-            compressFile(tmpOutputFilename, destname=outputFilename, remove=TRUE)
-
-        return(filterReport)
+        
+    } else {
+        stop("unknown file format: ", fileformat)
     }
+    
+    if(filecompr != "none")
+        compressFile(tmpOutputFilename, destname = outputFilename, remove = TRUE)
+    
+    return(filterReport)
+}
 
+#' @keywords internal
+#' @importFrom ShortRead readFasta writeFastq FastqStreamer yield narrow
+#' @importFrom Biostrings trimLRPatterns
+#' @importFrom S4Vectors evalSeparately
+#' @importFrom BiocGenerics width start end
```

Comment:
I think `BiocGenerics::width` is not used in `preprocessPairedReads` (trimming is not supported for paired reads)

Reply:
Correct (also start and end were not used). Fixed.